### PR TITLE
work in progress

### PR DESCRIPTION
**Pull request by r17x**
- [ ] [**Map**](https://docs.mapbox.com/mapbox-gl-js/api/map/) (🏗)
  - [x] instance member (addControl,addControlWithOptions)
- [ ] [**Properties and Options**](https://docs.mapbox.com/mapbox-gl-js/api/properties/) (🏗)
- [ ] [**Markers and Control**](https://docs.mapbox.com/mapbox-gl-js/api/markers/) (🚧)
  - [x] Marker instance member 
  - [ ] Marker Event
  - [x] Popup instance member 
  - [ ] Popup Event
- [x] [**Geography and Geometry**](https://docs.mapbox.com/mapbox-gl-js/api/geography/) (🏗)
  - [x] LngLat instance member
  - [x] LngLat static member
  - [x] LngLatBounds instance member
  - [x] LngLatBounds static member
  - [x] LngLatLike 
  - [x] LngLatBoundsLike
  - [x] Point
  - [x] PointLike
  - [x] MercatorCoordinate instance member
  - [x] MercatorCoordinate static member
  - [x] EdgeInsets static member (1/3)
- [ ] [**User interaction handlers**](https://docs.mapbox.com/mapbox-gl-js/api/handlers/) (🚧)
- [ ] [**Sources**](https://docs.mapbox.com/mapbox-gl-js/api/sources/) (🚧)
- [ ] [**Events**](https://docs.mapbox.com/mapbox-gl-js/api/events/) (🚧)

### CODE REVIEW

File: example/App_Intip.res [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable import/no-webpack-loader-syntax */
```
Comment (r17x) [inline example/App_Intip.res:1]: @adiatma can you try this and give me some feedback ? 

if you want to try, clone this or fork and checkout `dev`, then `yarn add path_to_clone_dir` in your project.

Comment (adiatma) [reply example/App_Intip.res:1]: Hi, bro @ri7nz thanks for mentioning me in here,

After [fork](https://github.com/adiatma/rescript-mapbox-gl) this repo, and clone to my computer, then trying to `yarn add <path/rescript-mapbox-gl>`, and add dependencies to `bsconfig.json` at `intip-chrome-extensions`. but I found this error after implementing it to replace  [https://github.com/adiatma/intip-chrome-extensions/blob/master/src/Map.res](https://github.com/adiatma/intip-chrome-extensions/blob/master/src/Map.res).

```bash
>>>> Start compiling
bsb: [2/6] src/Map.d
FAILED: src/Map.d
FAILED: Map has a self cycle
FAILED: cannot make progress due to previous errors.
>>>> Finish compiling(exit: 1)
```

thanks bro @ri7nz

Comment (r17x) [reply example/App_Intip.res:1]: Cause in `Mapbox` have module named `Map` and your implementation is `Map` that's make dependency cycle cause `Map` call it self.

image
image

It's work when your change `Map.res` to another name (e.g: `MapIntip.res`) and called `<MapIntip .../>`

Thank you @adiatma

Comment (adiatma) [reply example/App_Intip.res:1]: I see, thanks bro

**Pull request merged**